src/components/ContinuationIn/AuthorizationInformation.vue

```diff
@@ -285,14 +285,28 @@ export default class AuthorizationInformation extends Mixins(DateMixin, Document
     if (!documentKey || !documentName) return // safety check
 
     this.isDownloading = true
-    await this.downloadDocument(documentKey, documentName).catch(error => {
+    const documentClass = 'CORP'
+    try {
+      const docUrl: string = await this.getDownloadLink(documentKey, documentName, documentClass)
+      const link = document.createElement('a')
+      link.href = docUrl
+      link.download = documentName
+      link.target = '_blank' // This opens the link in a new browser tab
+
+      // Append to the document and trigger the download
+      document.body.appendChild(link)
+      link.click()
+
+      // Remove the link after the download is triggered
+      document.body.removeChild(link)
+      this.isDownloading = false
+    } catch (error) {
```

Comment:
In document-mixin.ts, there is existing code that looks a lot like the above...

Could the above block be moved (in its entirely) to document-mixin.ts? Whatever function you call it could throw an error that you would catch below (same as now).

Thoughts?

Reply:
Thanks for the refactor. Looks waaaay cleaner!

PS Let me resolve the conversations that I started. I use unresolved conversations to remember what to re-review. If you close them on me then I need to re-review everything (or look at all resolved convos). Thanks.